File: práctica 3/cola.py
class Cola:
    def _init_(self, n):
        self._arreglo = [0] * n  # Lista de tamaño fijo
        self._inicio = 0
        self._fin = -1
        self._n = n
        self._cantidad = 0

    def insert(self, e):
        if not self.isFull():
            self._fin = (self._fin + 1) % self._n
            self._arreglo[self._fin] = e
            self._cantidad += 1

    def remove(self):
        if not self.isEmpty():
            valor = self._arreglo[self._inicio]
            self._inicio = (self._inicio + 1) % self._n
            self._cantidad -= 1
            return valor
        return None  # Si está vacía

    def peek(self):
        if not self.isEmpty():
            return self._arreglo[self._inicio]
        return None  # Si está vacía

    def isEmpty(self):
        return self._cantidad == 0

    def isFull(self):
        return self._cantidad == self._n

    def size(self):
        return self._cantidad

# Ejemplo de uso
cola = Cola(5)
cola.insert(10)
cola.insert(20)
print("Frente de la cola:", cola.peek()) 
print("Elemento removido:", cola.remove()) 

File: práctica 3/cola.java
class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;
    private int cantidad;

    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = -1;
        this.cantidad = 0;
    }

    public void insert(long e) {
        if (!isFull()) {
            fin = (fin + 1) % n;
            arreglo[fin] = e;
            cantidad++;
        }
    }

    public Long remove() {
        if (!isEmpty()) {
            long valor = arreglo[inicio];
            inicio = (inicio + 1) % n;
            cantidad--;
            return valor;
        }
        return null; // Si está vacía
    }

    public Long peek() {
        if (!isEmpty()) {
            return arreglo[inicio];
        }
        return null; // Si está vacía
    }

    public boolean isEmpty() {
        return cantidad == 0;
    }

    public boolean isFull() {
        return cantidad == n;
    }

    public int size() {
        return cantidad;
    }
}

public class Main {
    public static void main(String[] args) {
        Cola cola = new Cola(5);
        cola.insert(10);
        cola.insert(20);
        System.out.println("Frente de la cola: " + cola.peek()); // Salida: 10
        System.out.println("Elemento removido: " + cola.remove()); // Salida: 10
    }
}